web/src/stream_settings_ui.ts

```diff
@@ -711,6 +805,7 @@ function setup_page(callback: () => void): void {
         const new_stream_announcements_stream_sub = stream_data.get_sub_by_name(
             new_stream_announcements_stream,
         );
+        const realm_has_archived_channels = stream_data.get_archived_subs().length > 0;
```

Comment:
For groups, filter is now shown irrespective of whether there are any deactivated groups or not. Not sure we want the same behavior here as well.

Reply:
Hm, how did we decide to  always show it for groups?

I was thinking a lot of realms would have no archived channels, and it adds unnecessary complexity to show the filter then. The same logic would probably apply to groups.

Reply:
I guess more precisely, it should be based on whether there are any archived channels the user has metadata access to.